core/opendaq/signal/include/opendaq/data_packet_impl.h

```diff
@@ -371,68 +371,100 @@ inline BaseObjectPtr DataPacketImpl<TInterface>::dataToObj(void* addr, const Sam
 }
 
 template <typename TInterface>
-inline StructPtr DataPacketImpl<TInterface>::buildStructFromPacket(void* addr,
-                                                                   const DataDescriptorPtr& descriptor,
-                                                                   const TypeManagerPtr& typeManager) const
+inline BaseObjectPtr DataPacketImpl<TInterface>::buildFromDescriptor(void*& addr,
+                                                                     const DataDescriptorPtr& descriptor,
+                                                                     const TypeManagerPtr& typeManager) const
 {
-    const auto name = descriptor.getName();
-    StructTypePtr structType = typeManager.getType(name);
+    const auto dimensions = descriptor.getDimensions();
+    const auto dimensionCount = dimensions.getCount();
 
-    const auto builder = StructBuilder(name, typeManager);
+    if (dimensionCount > 1)
+        throw NotSupportedException();
```

Comment:
This should provide an error message stating that a maximum of 1 dimension is currently supported:

```
throw NotSupportedException{"getLastValue on packets with dimensions supports only up to one dimension."};
```

Or something similar.